assignment02c/src/main/java/SmartHome.java

```diff
@@ -0,0 +1,126 @@
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+
+/**
+ * SmartHome is the container for all the Appliances and is responsible for issuing the commands to them.
+ */
+public class SmartHome {
+    
+    //list of appliances
+    private ArrayList<Appliance> appliances;
+    
+    private Thread controlThread;
+    private Runnable task;
+    
+    //primary appliances
+    private WaterHeater waterHeater;
+    private AirConditioner airConditioner;
+    private CookingOven cookingOven;
+    
+    //The Schdule of events to be carried out
+    private ArrayList<Event> schedule;
+    
+    
+    public SmartHome() {
+        appliances= new ArrayList<Appliance>();
+        
+        waterHeater = new WaterHeater();
+        airConditioner = new AirConditioner();
+        cookingOven = new CookingOven();
+        
+        appliances.add(waterHeater);
+        appliances.add(airConditioner);
+        appliances.add(cookingOven);
+        
+        schedule = new ArrayList<Event>();
+    }
+    
+    /**
+     * This method reads the schedule decided for the SmartHome from a file and parses it as a List of events
+     */
+    public void readSchedule(String filename) throws BadScheduleException, ScheduleNotFoundException {
+        BufferedReader br = null;
+        try {
+            String line;
+            br = new BufferedReader(new FileReader(filename));
+            while ((line = br.readLine()) != null) {
+				Event event = Event.parseEvent(line,this);
+//				System.out.println(line + " " + event + " " + event.getAppliance() + " " + getWaterHeater());
+				if(event != null) {
+				    schedule.add(event);
+				    if(event.getAppliance().canAutoSwitchOff() && event.getState() == true) {
+				        Event stopEvent = new Event(event.getAppliance(),event.getTime()+event.getAppliance().autoSwitchOffTime(), false);
```

Comment:
This is only supposed to be done if the appliance is not explicitly switched off.